zeebe/protocol/src/main/java/io/camunda/zeebe/protocol/record/Record.java

```diff
@@ -129,6 +129,9 @@ public interface Record<T extends RecordValue> extends JsonSerializable {
    */
   T getValue();
 
+  /**
+   * @return the id of the request that produced this record
```

Comment:
🔧 Let's enrich this explanation as @korthout asked https://github.com/camunda/zeebe/pull/17877#discussion_r1586311562. For example, we can mention the request id is unique within the broker. Therefore, the investigation you did to find out its uniqueness will buy us time later on.